apps/examples/wifi_manager/wm_test/wm_test_connectbyrssi_test.c

```diff
@@ -0,0 +1,311 @@
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	_wt_sta_connected,
+	_wt_sta_disconnected,
+	NULL, 
+	NULL,
+	_wt_scan_done,
+};
+
+/* Supported security method */
+static const char *g_wifi_test_auth_method[] = {
+#ifdef WT_AUTH_TABLE
+#undef WT_AUTH_TABLE
+#endif
+#define WT_AUTH_TABLE(type, str, desc) str,
+#include "wm_test_auth_table.h"
+};
+
+static const char *g_wifi_test_crypto_method[] = {
+#ifdef WT_CRYPTO_TABLE
+#undef WT_CRYPTO_TABLE
+#endif
+#define WT_CRYPTO_TABLE(type, str, desc) str,
+#include "wm_test_crypto_table.h"
+};
+
+void _wt_sta_connected(wifi_manager_cb_msg_s msg, void* arg)
+{
+	WT_LOG(TAG, "-->res(%d)", msg.res);		
+	WT_LOG(TAG, "bssid %02x:%02x:%02x:%02x:%02x:%02x",
+		msg.bssid[0], msg.bssid[1],
+		msg.bssid[2], msg.bssid[3],
+		msg.bssid[4], msg.bssid[5]);
+
+	snprintf(connected_ap_mac, sizeof(connected_ap_mac), "%02x:%02x:%02x:%02x:%02x:%02x",
+		msg.bssid[0], msg.bssid[1],
+		msg.bssid[2], msg.bssid[3],
+		msg.bssid[4], msg.bssid[5]);	
+
+	if (WIFI_MANAGER_SUCCESS == msg.res) {
+		WT_LOG(TAG, "Station Connected");
+	} else {
+		WT_LOGE(TAG, "Station not Connected");
+	}
+	WIO_TEST_SIGNAL;
+}
+
+void _wt_sta_disconnected(wifi_manager_cb_msg_s msg, void* arg)
+{	
+	WT_LOG(TAG, "--> res(%d) reason %d", msg.res, msg.reason);
+	WIO_TEST_SIGNAL;
+}
+
+void _wt_scan_done(wifi_manager_cb_msg_s msg, void* arg)
+{
+	WT_LOG(TAG, "-->");
+	/* Make sure you copy the scan results onto a local data structure.
+	 * It will be deleted soon eventually as you exit this function.
+	 */
+	if (msg.res != WIFI_MANAGER_SUCCESS || msg.scanlist == NULL) {
+		WIO_TEST_SIGNAL;
+		return;
+	}
+	_wt_print_scanlist(msg.scanlist);
+	wifi_manager_scan_info_s* wifi_scan_iter = msg.scanlist;
+	while (wifi_scan_iter != NULL) {
+		if (strncmp(wifi_scan_iter->ssid, opt->ssid, WIFIMGR_SSID_LEN) == 0) {
+			memcpy(bssid, wifi_scan_iter->bssid, strlen(wifi_scan_iter->bssid));
+			break;
+		}
+		wifi_scan_iter = wifi_scan_iter->next;
+	}
+	WIO_TEST_SIGNAL;
+}
+
+
+void _wt_print_scanlist(wifi_manager_scan_info_s *slist)
+{
+	while (slist != NULL) {		
+		WT_LOGP(TAG, "WiFi AP SSID: %-25s, BSSID: %-20s, Rssi: %d, Auth: %s, Crypto: %s\n",
+			   slist->ssid, slist->bssid, slist->rssi,
+			   g_wifi_test_auth_method[slist->ap_auth_type],
+			   g_wifi_test_crypto_method[slist->ap_crypto_type]);
+		slist = slist->next;
+	}
+}
+
+static int wm_connect_by_rssi_init(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = WIFI_MANAGER_SUCCESS;
+	res = wifi_manager_init(&g_wifi_callbacks);
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "Failed to init wifi_manager");
+		return 1 ;
+	}
+	WT_LEAVE;
+	return 0;
+}
+
+static void wm_connect_by_rssi_deinit(void)
+{
+	WT_ENTER;
+	wifi_manager_result_e res = wifi_manager_deinit();
+	if (res != WIFI_MANAGER_SUCCESS) {
+		WT_LOGE(TAG, "WiFi Manager failed to stop");
+	}	
+	WT_LEAVE;
+}
+
+static void _wt_print_wifi_ap_profile(wifi_manager_ap_config_s *config, char *title)
```

Comment:
please check can we use utils.c::wt_print_wifi_ap_profile

Reply:
Done